test/Common/MoistThermodynamics/profiles.jl

```diff
@@ -0,0 +1,124 @@
+#### Tested moist thermodynamic profiles
+
+#=
+This file contains functions to compute all of the
+thermodynamic _states_ that MoistThermodynamics is
+tested with in runtests.jl
+=#
+
+
+"""
+    fixed_lapse_rate(
+        param_set::AbstractParameterSet,
+        z::FT,
+        T_surface::FT,
+        T_min::FT,
+        ) where {FT <: AbstractFloat}
+
+Fixed lapse rate hydrostatic reference state
+"""
+function fixed_lapse_rate(
+    param_set::AbstractParameterSet,
+    z::FT,
+    T_surface::FT,
+    T_min::FT,
+) where {FT <: AbstractFloat}
+    _grav::FT = grav(param_set)
+    _cp_d::FT = cp_d(param_set)
+    _R_d::FT = R_d(param_set)
+    _MSLP::FT = MSLP(param_set)
+    Γ = _grav / _cp_d
+    z_tropopause = (T_surface - T_min) / Γ
+    H_min = _R_d * T_min / _grav
+    T = max(T_surface - Γ * z, T_min)
+    p = _MSLP * (T / T_surface)^(_grav / (_R_d * Γ))
+    T == T_min && (p = p * exp(-(z - z_tropopause) / H_min))
+    ρ = p / (_R_d * T)
+    return T, p, ρ
+end
+
+"""
+    tested_profiles(param_set, n::Int, ::Type{FT})
+
+A range of input arguments to thermodynamic state constructors
+
+ - `param_set` an `AbstractParameterSet`, see the [`MoistThermodynamics`](@ref) for more details
+ - `z_all` altitude
+ - `e_int` internal energy
+ - `ρ` (moist-)air density
+ - `q_tot` total specific humidity
+ - `q_pt` phase partition
+ - `T` air temperature
+ - `θ_liq_ice` liquid-ice potential temperature
+
+that are tested for convergence in saturation adjustment.
+
+Note that the output vectors are of size ``n*n_RS``, and they
+should span the input arguments to all of the constructors.
+"""
+function tested_profiles(
+    param_set::AbstractParameterSet,
+    n::Int,
+    ::Type{FT},
+) where {FT}
+
+    n_RS1 = 10
+    n_RS2 = 20
+    n_RS = n_RS1 + n_RS2
+    z_range = range(FT(0), stop = FT(2.5e4), length = n)
+    relative_sat1 = range(FT(0), stop = FT(1), length = n_RS1)
+    relative_sat2 = range(FT(1), stop = FT(1.02), length = n_RS2)
+    relative_sat = [relative_sat1..., relative_sat2...]
+    T_min = FT(150)
+    T_surface = FT(350)
```

Comment:
```suggestion
    T_surface = FT(300)
```
(Or why so high a surface temperature for testing?)

Reply:
I'm not sure, I didn't change what we originally had 🤷‍♂️. This PR was mostly to organize things and fix a bug in the profiles. I didn't intend for it to become complicated. I'll have to check to see if changing this value still passes moist thermo's test suite.

Reply:
Right, but the longstanding problem had been that the tests were thermodynamically inconsistent, which had led to problems in the past (e.g., issue #882 ). We had talked about fixing this several months ago, and this seems to be a good opportunity to do so.

Reply:
I'd prefer to keep this PR to do the few things mentioned in the description. There are several moist thermo PRs open right now, and I think these changes are better to be incorporated into #1059

Reply:
Ok. Fine with me. But it is important that we take care of the long-simmering substantial issues.